src/serial.rs

```diff
@@ -101,7 +140,15 @@ macro_rules! hal {
         ),
     )+) => {
         $(
+            $(#[$meta])*
+            /// The behaviour of the functions is equal for all three USARTs.
+            /// Except that they are using the corresponding USART hardware and pins.
             impl<PINS> Serial<$USARTX, PINS> {
+
+                /// Configures the Serial interface and creates an interface struct.
+                /// `baud_rate` configures the baud rate of the interface. The other parameters are
+                /// handles to the corresponding registers which must be changed by this function.
+                /// This function takes ownership of these registers.
```

Comment:
This isn't 100% correct for two reasons:

- Not all other parameters are registers, `Clocks` is a struct which 1. indicates that the clock configuration is frozen and 2, gives the frequencies of the various clocks. Also, pins are not registers, they are input and output pins
- It only takes ownership of the pins and usart device registers, the other registers are passed by reference in order to initialise the device

Reply:
Thanks for the clarification. I adapted it accordingly to your explanation and I learned a lot about the functionality :+1:

Reply:
Looks almost perfect now!

> The `serial` struct takes ownership of the `PINS` and `USARTX` device registers.

I read this as "takes ownership of the `PINS` device registers and `USARTX` device registers which is incorrect, `PINS` is a tuple of individual pins. Perhaps changing it to "Takes ownership over the `USARTX` device registers and the specified `PINS`" would make it more clear.

> The `serial` struct does not take ownership of `MAPR` and `APBX`

Another, more style related nitpick is that I'm not sure if this comment is required as that can be infered from them being passed as references. 

Other than those two things, it looks good to me

Reply:
As you wish, master!